File: server-scala-play-server/app/api/PostalCodeApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.PostalCodeResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait PostalCodeApi {
  /**
    * Create Postal Code
    * Create a Postal Code
    * @param accountId the id of the logged in user
    * @param code the postal code
    * @param latitude the latitude of the postal code
    * @param longitude the longitude of the postal code
    * @param stateCode the state code (e.g WA, WV, WI) that the postal code is located
    * @param city the city that the postal code is located
    * @param active whether the postal code created should be active or inactive
    */
  def createPostalCode(version: BigDecimal, accountId: Long, code: String, latitude: Double, longitude: Double, stateCode: Option[String], city: Option[String], active: Option[Boolean]): PostalCodeResponse

  /**
    * Delete Postal Code
    * Delete a Postal Code
    * @param accountId the id of the logged in user
    * @param postalCodeId the id of the postal code to delete
    */
  def deletePostalCode(version: BigDecimal, accountId: Long, postalCodeId: Long): SirqulResponse

  /**
    * Get Postal Code
    * Get a Postal Code
    * @param postalCodeId the id of the postal code to get
    */
  def getPostalCode(version: BigDecimal, postalCodeId: Long): PostalCodeResponse

  /**
    * Search Postal Codes
    * Get the list of regions. If latitude or longitude is null, will return all postal codes in the system with paginated response.
    * @param sortField the field to sort the results on
    * @param descending whether to order results in ascending or descending order
    * @param latitude the latitude of the postal code to search on
    * @param longitude the longitude of the postal code to search on
    * @param keyword the keyword of the postal code to search on
    * @param miles how far (in miles) to search on for the postal code
    * @param start the start of the index and/or pagination
    * @param limit the limit of the index and/or pagination
    */
  def getPostalCodes(version: BigDecimal, sortField: String, descending: Boolean, latitude: Option[Double], longitude: Option[Double], keyword: Option[String], miles: Option[Double], start: Option[Int], limit: Option[Int]): List[PostalCodeResponse]

  /**
    * Update Postal Code
    * Update a Postal Code
    * @param accountId the id of the logged in user
    * @param postalCodeId the id of the postal code to update
    * @param code the postal code to update
    * @param latitude the latitude of the postal code
    * @param longitude the longitude of the postal code
    * @param stateCode the state code (e.g. WA, WI, WY) where the postal code is located
    * @param city the city where the postal code is located
    * @param active whether the postal code is active or inactive
    */
  def updatePostalCode(version: BigDecimal, accountId: Long, postalCodeId: Long, code: Option[String], latitude: Option[Double], longitude: Option[Double], stateCode: Option[String], city: Option[String], active: Option[Boolean]): PostalCodeResponse
}

File: server-ruby-sinatra/api/postal_code_api.rb
require 'json'


MyApp.add_route('POST', '/api/3.18/postalCode/create', {
  "resourcePath" => "/PostalCode",
  "summary" => "Create Postal Code",
  "nickname" => "create_postal_code",
  "responseClass" => "PostalCodeResponse",
  "endpoint" => "/postalCode/create",
  "notes" => "Create a Postal Code",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "code",
      "description" => "the postal code",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "the latitude of the postal code",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "the longitude of the postal code",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "state_code",
      "description" => "the state code (e.g WA, WV, WI) that the postal code is located",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "city",
      "description" => "the city that the postal code is located",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "whether the postal code created should be active or inactive",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/postalCode/delete', {
  "resourcePath" => "/PostalCode",
  "summary" => "Delete Postal Code",
  "nickname" => "delete_postal_code",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/postalCode/delete",
  "notes" => "Delete a Postal Code",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "postal_code_id",
      "description" => "the id of the postal code to delete",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/postalCode/get', {
  "resourcePath" => "/PostalCode",
  "summary" => "Get Postal Code",
  "nickname" => "get_postal_code",
  "responseClass" => "PostalCodeResponse",
  "endpoint" => "/postalCode/get",
  "notes" => "Get a Postal Code",
  "parameters" => [
    {
      "name" => "postal_code_id",
      "description" => "the id of the postal code to get",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/postalCode/search', {
  "resourcePath" => "/PostalCode",
  "summary" => "Search Postal Codes",
  "nickname" => "get_postal_codes",
  "responseClass" => "Array<PostalCodeResponse>",
  "endpoint" => "/postalCode/search",
  "notes" => "Get the list of regions. If latitude or longitude is null, will return all postal codes in the system with paginated response.",
  "parameters" => [
    {
      "name" => "latitude",
      "description" => "the latitude of the postal code to search on",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "the longitude of the postal code to search on",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "the keyword of the postal code to search on",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "miles",
      "description" => "how far (in miles) to search on for the postal code",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the start of the index and/or pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "the limit of the index and/or pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "the field to sort the results on",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "whether to order results in ascending or descending order",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/postalCode/update', {
  "resourcePath" => "/PostalCode",
  "summary" => "Update Postal Code",
  "nickname" => "update_postal_code",
  "responseClass" => "PostalCodeResponse",
  "endpoint" => "/postalCode/update",
  "notes" => "Update a Postal Code",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "postal_code_id",
      "description" => "the id of the postal code to update",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "code",
      "description" => "the postal code to update",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "the latitude of the postal code",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "the longitude of the postal code",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "state_code",
      "description" => "the state code (e.g. WA, WI, WY) where the postal code is located",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "city",
      "description" => "the city where the postal code is located",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "whether the postal code is active or inactive",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

